Fps navigation mode followup

nyalldawson opened the pull request:
Some fixes and improvements to the FPS navigation mode

nyalldawson committed:
In fly mode, ctrl = slow down, shift = run

nyalldawson committed:
Rename "fly" navigation mode as "walk" mode

Fly has a different meaning here, where motion has inertia and the
key presses affect the acceleration of motion instead of directly
move the camera.

nyalldawson committed:
Cleanup some code, remove some unreachable conditions

nyalldawson committed:
Improve capturing of cursor in mouse lock mode

- Avoid jumpy first motion after activating lock
- Fix capture of cursor to map center -- because the camera controller
has no knowledge of the map canvas geometry (specifically the window
position), we can't set the global mouse cursor position directly from
the controller and instead need to push reponsibility for this up to
the 3d map canvas widget instead

uclaros commented:
`walk-mode` implies being at ground level and that could be a great useful third mode of camera control like you proposed on the original PR. Fly mode it is, fly like a drone though, not like a plane.

nyalldawson commented:
> walk-mode implies being at ground level and that could be a great useful third mode of camera control like you proposed on the original PR.

That's walk with gravity mode enabled :) (again, I'm copying Blender terminology here -- I think there's a large enough overlap of QGIS + Blender users to justify keeping consistent terminology and behavior between the applications.)

nyalldawson committed:
[3d] Add general 3d options tab, with settings for default camera
projection types and movement options

These settings are inherited by all new 3d views

nyalldawson committed:
Add global 3d setting to control whether vertical mouse axis movements should be inverted

nyalldawson committed:
Make elevation changes less sensitive vs ground movements changes

nyalldawson committed:
Remember camera movement speed changes triggered through fps nav mode
